File: 1444-NumberOfStepsToReduceANumberToZero/1444-NumberOfStepsToReduceANumberToZero.js
// Last updated: 3/6/2026, 2:08:58 PM
/**
 * @param {number} num
 * @return {number}
 */
var numberOfSteps = function(num) {
    let steps = 0;
    
    while(num>0){
        if(num%2===0){
            num=num/2;
            steps++;
        }
        else{
            num = num - 1;
            steps++;
        }
    }
    return steps;
};

File: 1444-NumberOfStepsToReduceANumberToZero/1444-NumberOfStepsToReduceANumberToZero.cpp
// Last updated: 7/17/2026, 4:26:28 PM
class Solution {
public:
    int numberOfSteps(int num) {
     int steps = 0;

     while(num > 0){
        if(num % 2 == 0){
            num = num / 2;
            steps++;
        }
        else{
            num = num - 1;
            steps++;
        }
     }
     return steps;
    }
};